File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numSet(nums.begin(),nums.end());
        int longestStreak=0;

        for(auto x:numSet){

            if(numSet.find(x-1)==numSet.end()){
                int currentStreak=1;
                int currNum=x;
                while(numSet.find(currNum+1)!=numSet.end()){
                    currentStreak++;
                    currNum++;
                    
                }
                longestStreak=max(longestStreak,currentStreak);
            }
        }
        return longestStreak;
    }
};

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int n = nums.length;
        int mid=0,i=0,j=n-1;
        if(n==1) return nums[0];
        while(i<=j){
            mid=i+(j-i)/2;

            if((mid==0||nums[mid]<nums[mid-1])
                &&(mid==n-1 || nums[mid]<nums[mid+1])
            ){
                return nums[mid];
            }else if(nums[mid]>nums[j]){
                i=mid+1;
            }else{
                j=mid-1;
            }
        }
        return nums[0];
    }
}
